waves/wallet/src/wallet.rs

```diff
@@ -285,14 +349,40 @@ impl Wallet {
     ///
     /// We choose to tag the derived key with `b"BLINDING_KEY"` in case we ever want to derive something else from the secret key.
     fn blinding_key(&self) -> SecretKey {
-        let h = hkdf::Hkdf::<sha2::Sha256>::new(None, self.secret_key.as_ref());
+        let h = Hkdf::<sha2::Sha256>::new(None, self.secret_key.as_ref());
 
         let mut bk = [0u8; 32];
         h.expand(b"BLINDING_KEY", &mut bk)
             .expect("output length aligns with sha256");
 
         SecretKey::from_slice(bk.as_ref()).expect("always a valid secret key")
     }
+
+    /// Derive the encryption key from the wallet's name and password.
+    ///
+    /// # Choice of salt
+    ///
+    /// The salt of HKDF can be public or secret and while it can operate without a salt, it is better to pass a salt value [0].
```

Comment:
:question: If we're gonna use a salt shouldn't it be random?

Reply:
If the salt is random, I need to store it on disk next to all the other data. I was thinking about but I wanted to store less data overall not more.

However, if the user deletes their browser storage, things are lost anyway ...

Reply:
I don't think it's very dangerous, but two users could have the same encryption key if they use the same name and password. Discovering the encryption key of one could make the other vulnerable, but of course our users are not in the same place so the risk is probably negligible.

Reply:
See my last commit, I am using a random salt now!